### ISSUE TITLE
Images of Imagebuttons not scaling

### ISSUE THREAD

**Issue by pitaxed** [prime31/Nez#814]
Not sure if this is a bug or not, but I was unable to scale/resize the Image of ImageButtons through any method on the Table, Row, Cell, the ImageButton or even the Image itself. I ended up changing the Layout() method of Image from 
`			imageWidth = size.X;
			imageHeight = size.Y;
to
`			imageWidth = size.X * scaleX;
			imageHeight = size.Y * scaleY;`
(Line 116 & 117)
and then it worked as expected. 
Is there another way I missed or should this be made into a pull request?

Also the overloaded constructor for ImageButtons taking three IDrawables suggests that the third IDrawable is to be displayed when the mouse hovers over it (ImageOver). But instead it gets set to the ImageChecked property, which I honestly don't know when it would be displayed, but it's certainly not what I expect or need.

**Comment by optimo** [comment]
I don't believe that's right. I have been able to place Images and have them resize generally speaking - depending on their parent container size and layout rules which might be limiting you in your experience. 

ImageButton, by default, has it's image's Scaling set to Scaling.Fit. just above lines 116 of Image.cs you can see the the scaling is applied on every layout; you may have been changing the size values, but there in Apply it would follow the rules for Fit where scale is calculated and new sizes returned. Whether that's exactly the hurdle you're up against, it's likely that or related to that.

for a general Image element, the scaling policy and size of the element would determine how large the image would get drawn (if it's parent table/group allows).

So what's a fix? well you could take the code from ImageButton class and make your own, or try subclassing it. but in doing that your own way you can decide another Scaling policy or do the math however you like. Maybe you set it to Scaling.Stretch (this is a default value for the enum, so you could also just not set scaling at all), and see if resizing the button also resizes the child image element. Subclassing or building your own elements for your UI needs is totally not weird, so consider it normal. Everyone's UI needs differ. The UI kit in nez in general does not resize a lot of things, Image element is sort of the exception because t figured out the hard sizes and draw routine use that. most of the other parts of ui can't be scaled so easily, and sometimes you have to cheat by rendering it at a lower resolution render texture for the compositing step.

The drawables thing seems like a typo resulting possibly from copy/past from another drawable and overlooked. (and could be fixed) It really should offer you the third argument with name reflecting that it's supposed to specify the ImageChecked drawable, and it does look that way as you saw following the constructors. There is some value in having a checked version of the image I suppose, and you could see it with SetChecked. But since you're more interested in the over/hover thing, you could use your custom class and instantiate the ImageButtonStyle with all the drawable elements you want or dont want, including the ImageOver drawable.

**Comment by davepruitt** [comment]
I am experiencing this same exact issue. My UI is fairly simple. The tree looks like this:

UICanvas is the top level node
UICanvas has a child that is a Table
Table has a child that is an ImageButton
The ImageButton is set to be top-right

This is the image I am using for the image button. It is 256x256:

![Image](https://github.com/user-attachments/assets/c6287305-2e89-4c72-b807-5b944b1bad51)

No matter what I do, that image button _will not resize_. I have tried calling `SetScale`, `SetSize`, `SetWidth`, and `SetHeight` on the ImageButton object. None of them have any effect.

**Comment by davepruitt** [comment]
I have found the bug in your code. So, the `ImageButton` class has a private `Image` member called `image`. This member `image` is set to an `IDrawable` for the various states - button-up, button-down, etc. etc. etc. This `image` member is then "added" as a child of the the `ImageButton` (which inherits from `Table` so it can have child elements).

When setting the scale of the `ImageButton`, this scale factor is never passed down to the `image` member, so it literally never even knows that it should be scaled. My temporary fix is to simply make `image` a public member and manually set the scale on _it_ rather than setting the scale on the `ImageButton` object.

So, in my temporary fix, my `ImageButton` class now begins like this:
```
	public class ImageButton : Button
	{
		public Image image;
```

And now when I set the scale in my code it looks like this:
```
var close_button = new ImageButton(close_button_up_sprite, close_button_down_sprite);
close_button.image.SetScale(0.25f);
```

This is a bit hackish so I think a better solution needs to be implemented, but that is what got it working for me.
